### PR TITLE
memory_hotplug: MEM_BOUNDARY_MB should be arch-dependent

### PR DESCRIPTION
**Pull request by Pennyzct**
As runtime/#624(https://github.com/kata-containers/runtime/pull/624#discussion_r212534586) discussed before, the size of memory section is arch-dependent.
For arm64, it should be 1G, not 128MB.

Fixes: #224

Signed-Off-By: Penny Zheng <penny.zheng@arm.com>

### CODE REVIEW

File: image-builder/image_builder.sh [hunk 1]
```diff
@@ -139,20 +144,21 @@ OK "Agent installed"
 ROOTFS_SIZE=$(du -B 1MB -s "${ROOTFS}" | awk '{print $1}')
 BLOCK_SIZE=${BLOCK_SIZE:-4096}
 OLD_IMG_SIZE=0
+OLD_MEM_BOUNDARY_MB=${MEM_BOUNDARY_MB}
```
Comment (grahamwhaley) [inline image-builder/image_builder.sh]: maybe s/OLD/ORIG/ - but, not a blocker.

File: image-builder/image_builder.sh [hunk 2]
```diff
@@ -50,8 +50,13 @@ MAX_IMG_SIZE_MB=2048
 
 FS_TYPE=${FS_TYPE:-"ext4"}
 
-# In order to support memory hotplug, image must be aligned to 128M
-MEM_BOUNDARY=128
+# In order to support memory hotplug, image must be aligned to memory section(size in MB) according to different architecture.
+ARCH=$(arch)
+case "$ARCH" in
+	aarch64)	MEM_BOUNDARY_MB=1024 ;;
+	 x86_64)	MEM_BOUNDARY_MB=128  ;;
```
Comment (grahamwhaley) [inline image-builder/image_builder.sh]: You could drop the x86_64 specialisation, as it is the same as the default ;-)

File: image-builder/image_builder.sh [hunk 3]
```diff
@@ -258,8 +263,8 @@ create_rootfs_disk()
 	# if the available disk space is less than rootfs size, repeat the process
 	# of disk creation by adding 5% in the inital assumed value $ROOTFS_SIZE
 	if [ $ROOTFS_SIZE -gt $AVAIL_DISK ]; then
-		# Increase the size but remain aligned to 128
-		MEM_BOUNDARY=$(($MEM_BOUNDARY+128))
+		# Increase the size but remain aligned to the original MEM_BOUNDARY_MB, which is stored in $OLD_MEM_BOUNDARY_MB
```
Comment (devimc) [inline image-builder/image_builder.sh]: s/OLD/ORG/

**Comment by Pennyzct** [comment]
Hi~ @devimc good catch, thanks! updated.;)

**Comment by jcvenegas** [comment]
LGTM

**Comment by jodh-intel** [comment]
lgtm

**Pull request merged**